### PR TITLE
Fix password recovery error logs

### PR DESCRIPTION
**Pull request by yathindrakodithuwakku**
### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- None

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/recovery-portal/src/main/webapp/challenge-question-add.jsp [hunk 1]
```diff
@@ -33,8 +33,9 @@
     String errorMsg = IdentityManagementEndpointUtil.getStringValue(request.getAttribute("errorMsg"));
 
     Map<String, List<ChallengeQuestionDTO>> challengeQuestionSets = new HashMap<String, List<ChallengeQuestionDTO>>();
+    String username = session.getAttribute("username");
 
-    if (session.getAttribute("username") != null) {
+    if (username != null) {
```
Comment (brionmario) [inline apps/recovery-portal/src/main/webapp/challenge-question-add.jsp:38]: Can the `username` be `""` here? If so, better to go for `StringUtils.isBlank`.

**Review by brionmario — APPROVED** [review]

**Review by brionmario — APPROVED** [review]

File: apps/recovery-portal/src/main/webapp/challenge-question-add.jsp [hunk 2]
```diff
@@ -33,8 +34,9 @@
     String errorMsg = IdentityManagementEndpointUtil.getStringValue(request.getAttribute("errorMsg"));
 
     Map<String, List<ChallengeQuestionDTO>> challengeQuestionSets = new HashMap<String, List<ChallengeQuestionDTO>>();
+    String username = session.getAttribute("username");
 
-    if (session.getAttribute("username") != null) {
+    if (!StringUtils.isBlank(username)) {
```
Comment (UdeshAthukorala) [inline apps/recovery-portal/src/main/webapp/challenge-question-add.jsp:39]: Can we use `StringUtils.isNotBlank()` method? Check other places also.

Comment (yathindrakodithuwakku) [reply apps/recovery-portal/src/main/webapp/challenge-question-add.jsp:39]: This has been fixed now.

Comment (yathindrakodithuwakku) [reply apps/recovery-portal/src/main/webapp/challenge-question-add.jsp:39]: Ref https://github.com/yathindrakodithuwakku/identity-apps/blob/6d819794409a7af3c175c492d6d42cc05b3d78f6/apps/recovery-portal/src/main/webapp/challenge-question-add.jsp#L39

**Pull request merged**